### PR TITLE
nmdc: add ADCGET, ADCSND, Direction, MaxedOut

### PR DESCRIPTION
**Pull request by aler9**
This patch adds all the file-related messages, documented in the reference paper:
http://nmdc.sourceforge.net/NMDC.html#_adcget
http://nmdc.sourceforge.net/NMDC.html#_adcsnd
... and so on

tests are provided as well

### CODE REVIEW

File: nmdc/files.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+)
+
+func init() {
+	RegisterMessage(&ADCGET{})
+	RegisterMessage(&ADCSND{})
+	RegisterMessage(&Direction{})
+	RegisterMessage(&MaxedOut{})
+}
+
+type ADCGET struct {
```
Comment (dennwc) [inline nmdc/files.go]:
```suggestion
type ADCGet struct {
```

File: nmdc/files.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+		if !bytes.Equal(data, []byte("ZL1")) {
+			return errors.New("ADCGET: invalid field 5")
+		}
+		m.Compressed = true
+	}
+
+	return nil
+}
+
+type ADCSND struct {
```
Comment (dennwc) [inline nmdc/files.go]:
```suggestion
type ADCSend struct {
```

**Review by dennwc — COMMENTED** [review]
Looks good, thanks! Only a small comment regarding the names:

**Comment by gswly** [comment]
this is ok

**Comment by aler9** [comment]
i renamed the messages and fixed unit tests

**Review by dennwc — APPROVED** [review]

**Pull request merged**